app/upgrades/v7/upgrades.go

```diff
@@ -77,6 +81,45 @@ func CreateUpgradeHandler(
 	}
 }
 
+// GetHostZone but with fixed historical stakeibc types
```

Comment:
yeah so let's say the stakeibc used to have schema A and now has schema B.

In the upgrade handler, it just imports the current type that's sitting in the repo. At the time that that code was written (and the time that the upgrade was run), this was type A.

But now we've just changed the repo's default type to B, so the old code can no longer compile. To make it more concrete, type A has UnbondingFrequency as an attribute, and type B does not. In the old upgrade handler, we try to access this attribute, which we can do successfully using type A, yet will error with type B.

Another way to think about this is from the perspective of the stride "application", it has no concept of history. This is synonymous with, say, import ibc-go v3 and then grabbing the memo field - it will cause a build error.

I'm not really sure what the best way to handle this in our codebase is. We technically don't need the upgrade handlers anymore since they'll never be run on this present binary, but I don't think we should remove them cause it's good to keep them as documentation, so I settled with modifying them just enough to where the original logic is preserved, but it can build and run tests successfully

Reply:
Sorry just realized you were asking about this function specifically. Per the reasoning above, GetHostZone will return the wrong type so I rewrote the function to return the correct type instead